jgrapht-core/src/main/java/org/jgrapht/alg/lca/NaiveLCAFinder.java

```diff
@@ -0,0 +1,266 @@
+ *
+ * <pre>
+ * 1. Start at each of nodes you wish to find the lca for (a and b)
+ * 2. Create sets aSet containing a, and bSet containing b
+ * 3. If either set intersects with the union of the other sets previous values (i.e. the set of notes visited) then
+ *    that intersection is LCA. if there are multiple intersections then the earliest one added is the LCA.
+ * 4. Repeat from step 3, with aSet now the parents of everything in aSet, and bSet the parents of everything in bSet
+ * 5. If there are no more parents to descend to then there is no LCA
+ * </pre>
+ *
+ * The rationale for this working is that in each iteration of the loop we are considering all the
+ * ancestors of a that have a path of length n back to a, where n is the depth of the recursion. The
+ * same is true of b.
+ *
+ * <p>
+ * We start by checking if a == b.<br>
+ * if not we look to see if there is any intersection between parents(a) and (parents(b) union b)
+ * (and the same with a and b swapped)<br>
+ * if not we look to see if there is any intersection between parents(parents(a)) and
+ * (parents(parents(b)) union parents(b) union b) (and the same with a and b swapped)<br>
+ * continues
+ *
+ * <p>
+ * This means at the end of recursion n, we know if there is an LCA that has a path of &lt;=n to a
+ * and b. Of course we may have to wait longer if the path to a is of length n, but the path to
+ * b&gt;n. at the first loop we have a path of 0 length from the nodes we are considering as LCA to
+ * their respective children which we wish to find the LCA for.
+ * 
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ *
+ * @author Leo Crawford
+ * @author Alexandru Valeanu
+ */
+public class NaiveLCAFinder<V, E> implements LCAAlgorithm<V>
+{
+    private Graph<V, E> graph;
+
+    /**
+     * Create a new instance of the naive LCA finder.
+     * 
+     * @param graph the input graph
+     */
+    public NaiveLCAFinder(Graph<V, E> graph) {
+        this.graph = Objects.requireNonNull(graph, "Graph cannot be null");
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public V getLCA(V a, V b)
+    {
+        if (!graph.vertexSet().contains(a))
+            throw new IllegalArgumentException("invalid vertex: " + a);
+
+        if (!graph.vertexSet().contains(b))
+            throw new IllegalArgumentException("invalid vertex: " + b);
+
+        return findLca(
+            Collections.singleton(a), Collections.singleton(b), new LinkedHashSet<>(),
+            new LinkedHashSet<>());
+    }
+
+    /**
+     * Return all the LCAs of a and b.
+     *
+     * @param a the first element to find LCA for
+     * @param b the other element to find the LCA for
+     *
+     * @return the set of all LCAs of a and b, or empty set if there is no LCA.
+     */
+    @SuppressWarnings("unchecked")
+    public Set<V> getAllLCAs(V a, V b)
+    {
+        Set<V>[] visitedSets = new Set[2];
```

Comment:
Since we are refactoring a little bit, improve this to the following

```
public Set<V> getAllLCAs(V a, V b) {
    @SuppressWarnings("unchecked")
    Set<V>[] visitedSets = (Set<V>[]) Array.newInstance(Set.class, 2);
```